pkg/scheduler/scheduler_test.go

```diff
@@ -0,0 +1,49 @@
+package scheduler
+
+import (
+	"testing"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	dynamicfake "k8s.io/client-go/dynamic/fake"
+	"k8s.io/client-go/kubernetes/fake"
+
+	karmadafake "github.com/karmada-io/karmada/pkg/generated/clientset/versioned/fake"
+)
+
+func TestCreateScheduler(t *testing.T) {
+	dynamicClient := dynamicfake.NewSimpleDynamicClient(runtime.NewScheme())
+	karmadaClient := karmadafake.NewSimpleClientset()
+	kubeClient := fake.NewSimpleClientset()
+
+	testcases := []struct {
+		name                     string
+		opts                     []Option
+		enableSchedulerEstimator bool
+	}{
+		{
+			name:                     "scheduler with default configuration",
+			opts:                     nil,
+			enableSchedulerEstimator: false,
+		},
+		{
+			name: "scheduler with enableSchedulerEstimator enabled",
+			opts: []Option{
+				WithEnableSchedulerEstimator(true),
```

Comment:
How about the `WithSchedulerEstimatorTimeout` and `WithSchedulerEstimatorPort` ? should we test them as well?

Reply:
I'll add tests for `WithSchedulerEstimatorPort`, but `Timeout` is hard to test unless we add a getter method. Considering `WithSchedulerEstimatorPort` somehow behaves the same as `WithSchedulerEstimatorTimeout`, I think it's ok.